app/(root)/layout.tsx

```diff
@@ -1,16 +1,23 @@
-import SideNavbar from '@/components/side-navbar/SideNavbar';
-import RightSidebar from '@/components/right-sidebar/RightSidebar';
-import React from 'react';
+import React from "react";
+import { User } from "next-auth";
 
-const Layout = ({ children }: { children: React.ReactNode }) => {
+import { auth } from "@/auth";
+import LeftNavbar from "@/components/left-navbar/LeftNavbar";
+import RightSidebar from "@/components/right-sidebar/RightSidebar";
+import { getUser } from "@/lib/actions/user.actions";
+
+const Layout = async ({ children }: { children: React.ReactNode }) => {
+  const session = await auth();
+  const userEmail = session && (await session.user?.email);
+  const user = userEmail && (await getUser(userEmail));
   return (
     <main>
-      <div className="flex h-screen justify-start">
-        <SideNavbar />
-        <section className="flex min-h-screen flex-1 flex-col px-6 pb-6 pt-36 max-md:pb-14 sm:px-14">
+      <div className="flex min-h-screen justify-start">
+        <LeftNavbar />
+        <section className="flex min-h-screen flex-1 flex-col pt-20 max-md:pb-14 sm:px-8">
           <div className="mx-auto w-full max-w-5xl">{children}</div>
```

Comment:
Is there a way to pass the user object to {children} ?

Reply:
"children" can be many components. So you can technically do this but it's a really fragile and prone-to-error way to do this. You could iterate over the children by turning the react children into an array and pass in user as props.

https://react.dev/reference/react/Children#children-toarray

I wouldn't recommend this. 

I wouldn't worry about sending the user to the children- We're passing it like this just because `RightSidebar` is a client-component, so we need to pass it down (or end up using state and useEffect and loading the information after the initial client-render). 

If we wanted to simplify this- can't `getUser` handle grabbing the session?

```js
  const session = await auth(); 
  const userEmail = session && (await session.user?.email);
  const user = userEmail && (await getUser(userEmail));
  
  // can become
  
  const user = await getUser();
  
  
  ```
```js
// user.actions.ts
async function _getUser() {
  const session = await auth();
  const email = session && (await session.user?.email);

  if (!email) return { error: "User not found!" };
  try {
    const user = await prisma.user.findUnique({
      where: {
        email,
      },
   ...etc....
   
 ```

Reply:
Oh interesting! Thank you!